### PR TITLE
dispmanx_vnc: initial addon

### PR DESCRIPTION
**Pull request by awiouy**
Hello!

Here is dispmanx_vnc and corresponding dependencies.

LE appears to ignore PKG_GIT_URL and PKG_GIT_BRANCH, so I used PKG_URL instead. Unfortunately, dispmanx_vnc does not provide release files, so I used master. Please tell me if this is acceptable, and if not, what I should do instead.

Also, I intend to move the configuration from the configuration file to the addon configuration GUI, as in, e.g., inadyn and syncthing.

Thank you for your attention, and regards, @

### CODE REVIEW

File: packages/addons/addon-depends/libvncserver/package.mk [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+################################################################################
+
+PKG_NAME="libvncserver"
+PKG_VERSION="0.9.10"
+PKG_REV="0"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE="https://libvnc.github.io/"
+PKG_URL="https://github.com/LibVNC/libvncserver/archive/LibVNCServer-$PKG_VERSION.tar.gz"
+PKG_DEPENDS_TARGET="toolchain libjpeg-turbo libpng"
```
Comment (vpeter4) [inline packages/addons/addon-depends/libvncserver/package.mk]: Add
`PKG_SOURCE_DIR="libvncserver-LibVNCServer-$PKG_VERSION"`
and remove unpack() function.

**Comment by MilhouseVH** [comment]
> Unfortunately, dispmanx_vnc does not provide release files, so I used master. Please tell me if this is acceptable, and if not, what I should do instead.

In that case use a revision, ie. `PKG_VERSION="78e6673"` and change PKG_URL to:
```
PKG_URL="https://github.com/patrikolausson/dispmanx_vnc/archive/$PKG_VERSION.tar.gz"
```

File: packages/addons/service/dispmanx_vnc/package.mk [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+PKG_IS_ADDON="yes"
+PKG_ADDON_TYPE="xbmc.service"
+PKG_ADDON_PROVIDES=""
+PKG_ADDON_PROJECTS="RPi RPi2"
+PKG_ADDON_REPOVERSION="8.0"
+
+PKG_AUTORECONF="no"
+
+PKG_MAINTAINER="Lukas Rusak (lrusak at irc.freenode.net)"
+PKG_MAINTAINER="Anton Voyl (awiouy at gmail dot com)"
```
Comment (MilhouseVH) [inline packages/addons/service/dispmanx_vnc/package.mk]: You can only have one `PKG_MAINTAINER`

Comment (awiouy) [inline packages/addons/service/dispmanx_vnc/package.mk]: @lrusak What do you prefer: You, me, somebody else, nobody?

Comment (chewitt) [inline packages/addons/service/dispmanx_vnc/package.mk]: If you're happy to be the maintainer? .. set it to you :)

File: packages/addons/service/dispmanx_vnc/source/system.d/service.system.dispmanx_vnc.service [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=dispmanx_vnc
+After=kodi.service
+
+[Service]
+ExecStart=/bin/sh /storage/.kodi/addons/service.system.dispmanx_vnc/bin/dispmanx_vncserver-service
```
Comment (lrusak) [inline packages/addons/service/dispmanx_vnc/source/system.d/service.system.dispmanx_vnc.service]: Should just be dispmanx_vncserver not sure where -service comes from

Comment (awiouy) [inline packages/addons/service/dispmanx_vnc/source/system.d/service.system.dispmanx_vnc.service]: Hi @lrusak 
I moved all pieces of script (to copy the default config, to wait) into dispmanx_vncserver-service shell script. This is also where I intend to collect configuration from the addon, in order to pass it on to the service (see, e.g., inadyn-service). This way, users will be able to configure the service from Kodi, without having to ssh to LE. I hope you agree with this.
Regards, @

Comment (lrusak) [inline packages/addons/service/dispmanx_vnc/source/system.d/service.system.dispmanx_vnc.service]: Ah yes sorry, I missed that

Comment (MilhouseVH) [inline packages/addons/service/dispmanx_vnc/source/system.d/service.system.dispmanx_vnc.service]: From a purely naming perspective, calling the wrapper shell script a service is rather confusing - why not just call it `dispmanx_vncserver.sh`?

Comment (awiouy) [inline packages/addons/service/dispmanx_vnc/source/system.d/service.system.dispmanx_vnc.service]: Hi @MilhouseVH
What's in a name? :D
I guess I got used to end the name of my service wrappers with -service (see also inadyn, syncthing and tinc). By the way, I have seen such scripts ending with -start. So I presume it is a matter of convention.
Regards, @

**Pull request merged**